telemetry/data_window.go

```diff
@@ -80,7 +113,128 @@ var (
+	result += sqlTypeDeta.Other
+	return result
+}
+
+func readSQLMetric(timepoint time.Time, SQLresult sqlUsageData) (sqlUsageData, error) {
+	ctx := context.TODO()
+	quantile := 0.99
+
+	result, err := querySQLMetric(ctx, timepoint, quantile)
+	if err != nil {
+		if err1, ok := err.(*promv1.Error); ok {
+			return SQLresult, errors.Errorf("query metric error, msg: %v, detail: %v", err1.Msg, err1.Detail)
+		}
+		return SQLresult, errors.Errorf("query metric error: %v", err.Error())
+	}
+
+	dataAnylis(result, &SQLresult)
+	return SQLresult, nil
+
+}
+
+func querySQLMetric(ctx context.Context, queryTime time.Time, quantile float64) (result pmodel.Value, err error) {
+	// Add retry to avoid network error.
+	var prometheusAddr string
+	for i := 0; i < 5; i++ {
+		//TODO: the prometheus will be Integrated into the PD, then we need to query the prometheus in PD directly, which need change the quire API
+		prometheusAddr, err = infosync.GetPrometheusAddr()
+		if err == nil || err == infosync.ErrPrometheusAddrIsNotSet {
+			break
+		}
+		time.Sleep(100 * time.Millisecond)
+	}
+	if err != nil {
+		return nil, err
+	}
+	promClient, err := api.NewClient(api.Config{
+		Address: prometheusAddr,
+	})
+	if err != nil {
+		return nil, err
+	}
+	promQLAPI := promv1.NewAPI(promClient)
+	ctx, cancel := context.WithTimeout(ctx, promReadTimeout)
+	defer cancel()
+	promQL := "sum(tidb_executor_statement_total{}) by (instance,type)"
+	// Add retry to avoid network error.
+	for i := 0; i < 5; i++ {
+		result, _, err = promQLAPI.Query(ctx, promQL, queryTime)
+		if err == nil {
+			break
+		}
+		time.Sleep(100 * time.Millisecond)
+	}
+	return result, err
+}
+
+func dataAnylis(promResult pmodel.Value, SQLresult *sqlUsageData) {
+	switch promResult.Type() {
+	case pmodel.ValVector:
+		matrix := promResult.(pmodel.Vector)
+		for _, m := range matrix {
+			v := m.Value
+			genRecord(m.Metric, v, SQLresult)
+		}
+	}
+}
+
+func genRecord(metric pmodel.Metric, pair pmodel.SampleValue, SQLresult *sqlUsageData) {
+	v := ""
+	if metric != nil {
+		v = string(metric[pmodel.LabelName("type")])
+	}
+	if len(v) == 0 {
+		v = "total"
+	}
+	var record int64
+	if math.IsNaN(float64(pair)) {
+		record = 0
+	} else {
+		record = int64(float64(pair))
+	}
+	switch v {
+	case "total":
+		SQLresult.SQLTotal = record
+	case "Use":
+		SQLresult.SQLType.Use = record
+	case "Show":
+		SQLresult.SQLType.Show = record
+	case "Begin":
+		SQLresult.SQLType.Begin = record
+	case "Commit":
+		SQLresult.SQLType.Commit = record
+	case "Rollback":
+		SQLresult.SQLType.Rollback = record
+	case "Insert":
+		SQLresult.SQLType.Insert = record
+	case "Replace":
+		SQLresult.SQLType.Replace = record
+	case "Delete":
+		SQLresult.SQLType.Delete = record
+	case "Update":
+		SQLresult.SQLType.Update = record
+	case "Select":
+		SQLresult.SQLType.Select = record
+	default:
+		SQLresult.SQLType.Other = record
+	}
+}
+
 // RotateSubWindow rotates the telemetry sub window.
+// ctx sessionctx.Context
```

Comment:
Remove useless comment

Reply:
Thanks for remingding! I have already fix the coding format in my latest upload.